src/job.rs

```diff
@@ -939,13 +939,50 @@ pub fn job_send_msg(context: &Context, msg_id: MsgId) -> Result<()> {
     Ok(())
 }
 
+fn add_imap_deletion_jobs(context: &Context) -> sql::Result<()> {
+    if let Some(delete_server_after) = context.get_config_delete_server_after() {
+        let threshold_timestamp = time() - delete_server_after;
+
+        // Select all expired messages which don't have a
+        // corresponding message deletion job yet.
+        let msg_ids = context.sql.query_map(
+            "SELECT id FROM msgs \
+             WHERE timestamp < ? \
+             AND server_uid != 0 \
+             AND NOT EXISTS (SELECT 1 FROM jobs WHERE foreign_id = msgs.id)",
```

Comment:
this currently also prevents a delete-job when other jobs are pending for msg.id (eg. a MarkseenMsgOnImap). not sure if this is big deal as the job would be queued later on, however, if this is on-purpose, maybe add this to the comment.

Reply:
I should check for job type, `foreign_id` may mean different things.